### PR TITLE
Source Salesloft: certify to beta

### PR DESCRIPTION
**Pull request by davydov-d**
## What
https://github.com/airbytehq/airbyte/issues/19217

## How
 - update docs
 - unit test coverage -> 99%
 - integration test strictness level -> high
 - add auth by api key
 - update stream schemas
 - add incremental sync support to more streams
 - add allowed hosts

### CODE REVIEW

**Comment by davydov-d** [comment]
> /test connector=connectors/source-salesloft

:point_up: 2 tests will fail because of https://github.com/airbytehq/airbyte/pull/23681

**Review by bazarnov — APPROVED** [review]
LGTM for code changes, tests are failing on backward compatibility checks, make sure this is covered, otherwise - ok.

**Review by lazebnyi — APPROVED** [review]

File: airbyte-config/init/src/main/resources/seed/source_definitions.yaml [hunk 1]
```diff
@@ -1727,6 +1727,9 @@
   icon: salesloft.svg
   sourceType: api
   releaseStage: alpha
+  allowedHosts:
```
Comment (bnchrch) [inline airbyte-config/init/src/main/resources/seed/source_definitions.yaml:1730]: 👍

**Review by bnchrch — APPROVED** [review]
Nothing stood out to me here.
1. good cursors
2. Good primary keys
3. Great testing and documentation
4. It likely is a breaking change but thats ok as its an alpha connector

**Pull request merged**